core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/util/CryptoUtil.java

```diff
@@ -242,6 +284,22 @@ public String encryptAndBase64Encode(byte[] plainText) throws
         return Base64.encode(encrypt(plainText));
     }
 
+    /**
+     * Encrypt the given plain text and base64 encode the encrypted content.
+     *
+     * @param plainText The plaintext value to be encrypted and base64
+     *                  encoded.
+     * @param secretKey The secret key to be used for encryption.
+     * @return The base64 encoded cipher text
+     * @throws CryptoException On error during encryption
+     */
+    public String encryptWithCustomKeyAndBase64Encode(byte[] plainText, String secretKey) throws
```

Comment:
Should be able to overload the `encryptAndBase64Encode` if we decide to overload the encrypt method